op-node/rollup/async/asyncgossiper.go

```diff
@@ -0,0 +1,123 @@
+package async
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+// AsyncGossiper is a component for gossiping a payload while running the derivation pipeline
+// the gossiping is done in a separate goroutine, so that the derivation pipeline can continue without waiting for the gossiping to finish
+// it also maintains the currently gossiping payload, so it can be checked against other found payloads
+type AsyncGossiper struct {
+	running atomic.Bool
+	// channel to add new payloads to gossip
+	set chan *eth.ExecutionPayload
+	// channel to request getting the currently gossiping payload
+	get chan chan *eth.ExecutionPayload
+	// channel to request clearing the currently gossiping payload
+	clear chan struct{}
+
+	hasPayload     atomic.Bool
+	currentPayload *eth.ExecutionPayload
+	net            Network
+}
+
+// To avoid import cycles, we define a new Network interface here
+// this interface is compatable with driver.Network
+type Network interface {
+	PublishL2Payload(ctx context.Context, payload *eth.ExecutionPayload) error
+}
+
+func NewAsyncGossiper(net Network) *AsyncGossiper {
+	return &AsyncGossiper{
+		running: atomic.Bool{},
+		set:     make(chan *eth.ExecutionPayload, 1),
+		get:     make(chan chan *eth.ExecutionPayload),
+		clear:   make(chan struct{}),
+
+		hasPayload:     atomic.Bool{},
+		currentPayload: nil,
+		net:            net,
+	}
+}
+
+// Gossip is an exposed, synchronous function to send a payload to be gossiped into the AsyncGossiper
+func (p *AsyncGossiper) Gossip(payload *eth.ExecutionPayload) {
+	// send the payload to the newPayloads channel. this will block until the payload can be sent
+	p.set <- payload
```

Comment:
This blocks until we can _start_ sending the payload right? It's not actually synchronous in that the block will have actually been published by the time it returns.